src/boundaries/stagger_and_step.jl

```diff
@@ -0,0 +1,255 @@
+export stagger_and_step
+using LinearAlgebra: norm
+using Random
+using ProgressMeter
+
+
+
+
+
+"""
+    stagger_and_step(ds::DynamicalSystem, x0, N::Int, 
+        isinside::Function; kwargs...) -> trajectory
+
+Implement the stagger-and-step method 
+[Sweet2001](@cite) and [Sala2016](@cite) to approximate the invariant 
+non-attracting set governing the chaotic transient dynamics 
+of a system, namely the stable manifold of a chaotic saddle. 
+
+Given the dynamical system `ds` and a initial guess `x0` in a 
+region *with no attractors* defined by the membership function 
+`isinside`, the algorithm returns a `StateSpaceState` named `trajectory` 
+of `N` points from the phase space close to the stable manifold. 
+If we set one of this point as an initial condition of `ds`, 
+the trajectory escape from the region after at least `Tm` 
+steps of `ds`. The search is stochastic and depends on the 
+parameter `δ` defining a (small) neighborhood of search. 
+
+The function `isinside(x)` must return `true` if the point `x` is 
+inside the chosen bounded region and `false` otherwise. See 
+[`statespace_sampler`](@ref) as an helper to construct this 
+function.
+
+## Keyword arguments
+* `δ = 1e-10`: A small number constraining the random
+  search around a particular point. The interpretation of this 
+  number will depend on the probability distribution chosen 
+  for the sampling (see `stagger_mode`).
+
+* `Tm = 30`: The minimum number of iterations of `ds` before 
+  the trajectory escapes from the bounding box defined by
+  `isinside`. 
```

Comment:
With respect to `T` and `Tm`: what do these mean for continuous time systems? In DynamicalSystems.jl we should try to have a consistent terminology. Let's not use the "iteration" word. Let's just use "steps". This directly refers to the `step!` function. Since this algorithm applies to continuous time systems as well, this function should also allow for the keywords `Δt` and `stop_at_Δt`, which are propagated to `step!`. Just like most other functions of the library.

Reply:
This also means that you should change all `step!` calls to `step!(ds, Δt, stop_at_Δt)`. By default `Δt = 1` and `stop = false`.

Reply:
Or, do you think it is better for continuous time systems to only use the integrator iteration? So if the actual amount of time the system evolves doesn't matter, then ignore my suggestion. But still simply change "iteration" to "step".

Reply:
Or, do you think it is better for continuous time systems to only use the integrator iteration? So if the actual amount of time the system evolves doesn't matter, then ignore my suggestion. But still simply change "iteration" to "step".